contracts/StoryDataSource.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "@chainlink/contracts/src/v0.8/ChainlinkClient.sol";
+import "@chainlink/contracts/src/v0.8/ConfirmedOwner.sol";
+import "./interfaces/IOffChainDataSource.sol";
+import { Strings } from '@openzeppelin/contracts/utils/Strings.sol';
+import "./interfaces/IStoryOracleCallback.sol";
+import "./libraries/StoryOracleClient.sol";
+import "./libraries/StoryOracleHelper.sol";
+
+contract StoryDataSource  is IOffChainDataSource, IStoryOracleCallback, StoryOracleClient {
+    using StoryOracleHelper for StoryOracleHelper.Request;
+    using Strings for uint256;
+
+    mapping(address=>string) public userData;
+    mapping(bytes32=>address) public requests;
+
+    event RequestData(bytes32 indexed requestId, string data);
+
+    constructor()  {}
+
+    function getData(address user) public view override returns(string memory data) {
```

Comment:
Wonder if we can use external instead of public here since getData is not used internally. Same for the other functions.

Reply:
Yep, we can use external here.